torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -16,6 +16,64 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+namespace {
+
+// Deterministic set implementation using vector of unique entries
+//  Tries to support the basic API's that std::unordered_set has
+// and provides a deterministic iteration order.
+// TODO:
+//  1. Both this class and VectorOfUniqueEntries should go into
+// a new ADT file.
+//  2. Should use this class for a lot of set/vec pair implementation
+// logic.
+template <typename T>
+class DeterministicSet : public VectorOfUniqueEntries<T> {
```

Comment:
Do we need another class? Can we just add the new methods to `VectorOfUniqueEntries`?

Reply:
I guess not. Just thought if we'd want to expand `VectorOfUniqueEntries` we'd benefit from a `set-like` interface and a `vector-like` interface, as it could be used as both a `vec` and a `set`. But trying to add std like interfaces of both `set` and `vec` in a single class could make it quite crowded. 

Not any concern for now. Just removed it. Thanks.